general_property_lists/general_property_lists.md

```diff
@@ -0,0 +1,239 @@
+
+# Properties for context, program, sampler, and accessor: Expanding use of property lists
+
+## Overview
+
+With the move towards a [generalizing SYCL](https://github.com/KhronosGroup/SYCL-Shared/blob/master/proposals/sycl_generalization.md),
+future backends may have use of parameters passed to their parent SYCL object.
+However, with only `buffer`, `image`, and `queue` exposing constructors with a
+`property_list` parameter, the extensibility of the parameter space of all other
+user-constructible SYCL objects is limited.
+
+This proposal extends the constructors for `context`, `program`, `sampler`, and
+`accessor` with an additional `property_list` parameter.
+
+## Specification changes
+
+Section 4.3.4 of the 1.2.1 SYCL specifications (Rev 6) is changed to mention
+`context`, `program`, `sampler`, and `accessor` as providing a `property_list`
+parameter in their constructors. This is in addition to the existing mention of
+`buffer`, `image`, and `queue`.
+
+## Interface changes
+
+The constructors for `context`, `program`, `sampler`, and `accessor`
+are extended to have a `property_list` parameter, similar to `buffer`, `image`, and
+`queue`.
+
+### Context changes
+
+```cpp
+namespace cl {
+namespace sycl {
+class context {
+public:
+    context(const property_list &propList = {});
+
+    explicit context(async_handler asyncHandler,
+        const property_list &propList = {});
+
+    context(const device &dev, const property_list &propList = {});
+
+    context(const device &dev, async_handler asyncHandler,
+        const property_list &propList = {});
+
+    context(const platform &plt, const property_list &propList = {});
+
+    context(const platform &plt, async_handler asyncHandler,
+        const property_list &propList = {});
+
+    context(const vector_class<device> &deviceList,
+        const property_list &propList = {});
+
+    context(const vector_class<device> &deviceList,
+        async_handler asyncHandler, const property_list &propList = {});
+
+    ...
+};
+} // namespace sycl
+} // namespace cl
+```
+
+### Program changes
+
+```cpp
+namespace cl {
+namespace sycl {
+class program {
+public:
+    explicit program(const context &context,
+        const property_list &propList = {});
+
+    program(const context &context, vector_class<device> deviceList,
+        const property_list &propList = {});
+
+    program(vector_class<program> &programList, string_class linkOptions = "",
+        const property_list &propList = {});
+
+    ...
+};
+} // namespace sycl
+} // namespace cl
+```
+
+### Sampler changes
+
+```cpp
+namespace cl {
+namespace sycl {
+class sampler {
+public:
+    sampler(coordinate_normalization_mode normalizationMode,
+        addressing_mode addressingMode, filterin_mode filteringMode,
+        const property_list &propList = {});
+
+    sampler(cl_sampler clSampler, const context &syclContext,
```

Comment:
We tend not to include overloads for the interop constructors with property_list as it's not always possible to enforce the properties.

Reply:
Good point. I have removed this constructor.